state-chain/chains/src/cf_parameters.rs

```diff
@@ -28,6 +39,40 @@ pub struct VaultSwapParameters {
 	pub affiliate_fees: BoundedVec<AffiliateAndFee, ConstU32<MAX_AFFILIATES>>,
 }
 
+/// Provide a function that builds and encodes `cf_parameters`.
+/// The return type is encoded Vec<u8>, which circumvents the difference in return types depending
+/// on if CCM data is available.
+pub fn build_cf_parameters(
+	refund_parameters: ChannelRefundParametersDecoded,
+	dca_parameters: Option<DcaParameters>,
+	boost_fee: u8,
+	broker_id: AccountId,
+	broker_commission: BasisPoints,
+	affiliate_fees: BoundedVec<AffiliateAndFee, ConstU32<MAX_AFFILIATES>>,
+	ccm: Option<&CcmChannelMetadata>,
+) -> Vec<u8> {
+	let vault_swap_parameters = VaultSwapParameters {
+		refund_params: refund_parameters,
+		dca_params: dca_parameters,
+		boost_fee,
+		broker_fee: Beneficiary { account: broker_id, bps: broker_commission },
+		affiliate_fees,
+	};
+
+	match ccm {
+		Some(ccm) => VersionedCcmCfParameters::V0(CfParameters {
+			ccm_additional_data: ccm.ccm_additional_data.clone(),
+			vault_swap_parameters,
+		})
+		.encode(),
+		None => VersionedCfParameters::V0(CfParameters {
+			ccm_additional_data: (),
+			vault_swap_parameters,
+		})
+		.encode(),
+	}
```

Comment:
```suggestion
	VersionedCcmCfParameters::V0(CfParameters {
		ccm_additional_data: ccm.map(|ccm| ccm.ccm_additional_data.clone()),
		vault_swap_parameters,
	})
	.encode()
```

Reply:
The CfParameters must have either Bytes, or () as its `ccm_additional_data` type (It's not an Option)
so we cannot dynamically map this into different return types within the same statement.

THis is also the reason why we return Encoded Bytes here, instead of the `VersionedCcmCfParameters` type, because the Generic will be different depending on whether CCM is present or not.